mobilecoind-json/src/main.rs

```diff
@@ -0,0 +1,317 @@
+
+    Ok(Json(JsonRequestResponse {
+        request_code: String::from(resp.get_b58_code()),
+    }))
+}
+
+#[derive(Deserialize)]
+struct JsonTransferRequest {
+    request_code: String,
+    amount: u64,
+}
+#[derive(Serialize)]
+struct JsonTransferResponse {
+    key_image: String,
+    tombstone: u64,
+}
+
+/// Performs a transfer from a monitor and subaddress. The target and amount are in the POST data.
+#[post(
+    "/monitors/<monitor_hex>/transfer/<index>",
+    format = "json",
+    data = "<transfer>"
+)]
+fn transfer(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    index: u64,
+    transfer: Json<JsonTransferRequest>,
+) -> Result<Json<JsonTransferResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    let mut req = mc_mobilecoind_api::ReadRequestCodeRequest::new();
+    req.set_b58_code(transfer.request_code.clone());
+    let resp = state
+        .mobilecoind_api_client
+        .read_request_code(&req)
+        .map_err(|err| format!("Failed reading request code: {}", err))?;
+    let public_address = resp.get_receiver();
+
+    let mut outlay = mc_mobilecoind_api::Outlay::new();
+    outlay.set_receiver(public_address.clone());
+    outlay.set_value(transfer.amount);
+
+    let mut req = mc_mobilecoind_api::SendPaymentRequest::new();
+    req.set_sender_monitor_id(monitor_id);
+    req.set_sender_subaddress(index);
+    req.set_outlay_list(RepeatedField::from_vec(vec![outlay]));
+
+    let resp = state
+        .mobilecoind_api_client
+        .send_payment(&req)
+        .map_err(|err| format!("Failed to send payment: {}", err))?;
+
+    let receipt = resp.get_sender_tx_receipt();
+    return Ok(Json(JsonTransferResponse {
+        key_image: hex::encode(receipt.get_key_image_list()[0].get_data()),
+        tombstone: receipt.get_tombstone(),
+    }));
+}
+
+#[derive(Serialize, Default)]
+struct JsonStatusResponse {
+    status: String,
+}
+
+/// Checks the status of a transfer given a key image and tombstone block
+#[get("/status/<key_hex>/<tombstone>")]
+fn status(
+    state: rocket::State<State>,
+    key_hex: String,
+    tombstone: u64,
+) -> Result<Json<JsonStatusResponse>, String> {
+    let mut receipt = mc_mobilecoind_api::SenderTxReceipt::new();
+    let mut key_image = KeyImage::new();
+    key_image.set_data(
+        hex::decode(key_hex).map_err(|err| format!("Failed to decode key image hex: {}", err))?,
+    );
+    receipt.set_key_image_list(RepeatedField::from_vec(vec![key_image]));
+    receipt.set_tombstone(tombstone);
+
+    let mut req = mc_mobilecoind_api::GetTxStatusAsSenderRequest::new();
+    req.set_receipt(receipt);
+
+    let resp = state
+        .mobilecoind_api_client
+        .get_tx_status_as_sender(&req)
+        .map_err(|err| format!("Failed getting status: {}", err))?;
+    let status = resp.get_status();
+    match status {
```

Comment:
You can save a bunch of repetitions here by mapping the status to a string and then returning `Ok(Json(JsonStatusResponse { status: status_str }))`.